document_page_approval/tests/test_document_page_approval.py

```diff
@@ -0,0 +1,275 @@
+from odoo import Command
+from odoo.exceptions import UserError
+from odoo.tests import new_test_user
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+class TestDocumentPageApproval(BaseCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.page_obj = cls.env["document.page"]
+        cls.history_obj = cls.env["document.page.history"]
+
+        # Demo Data
+        cls.category1 = cls.env.ref("document_page.demo_category1")
+        cls.page1 = cls.env.ref("document_page.demo_page1")
+
+        # Create test user without groups first
+        cls.user2 = new_test_user(cls.env, login="test-user2", groups="base.group_user")
+
+        # Ensure user2 has the approver group
+        cls.approver_gid = cls.env.ref(
+            "document_page_approval.group_document_approver_user"
+        )
+        cls.user2.write({"groups_id": [Command.link(cls.approver_gid.id)]})
```

Comment:
```suggestion
        cls.user2 = new_test_user(cls.env, login="test-user2", groups="document_page_approval.group_document_approver_user")

        cls.approver_gid = cls.env.ref(
            "document_page_approval.group_document_approver_user"
        )
```
The `base.group_user` permission is inherited indirectly with `document_knowledge`